data/components/main-toolbar.js

```diff
@@ -88,7 +89,8 @@ var MainToolbar = React.createClass({
         ),
         ToolbarButton({bsSize: "xsmall", onClick: this.onSwitchPerspective},
           perspectiveLabel
-        )
+        ),
+        ConnectionFilter(this.props)
```

Comment:
My guess is that we may rather put the connection filter on the left of the SearchBox. Otherwise, the appearance of the dropdown list moves the search box and may disturb the user.

Also I am a bit curious why the SearchBox is in `componentDidMount`. Ping @janodvarko?